### PR TITLE
Re-calculate Popover position on screen resize

### PR DESCRIPTION
**Pull request by Kemosabert**
### Description
Fix for issue #304. The issue here was that the throttle function did not work as expected due to how event pooling works in React, hence the callback function was rarely called. Fixed this 

### Breaking changes
none

### CODE REVIEW

File: components/popover/Popover.js [hunk 1]
```diff
@@ -23,11 +23,11 @@ const factory = (axis, calculatePositions, Overlay) => {
       },
     };
 
-    setPlacementThrottled = () => throttle(this.setPlacement, 16);
-
     componentDidMount() {
       document.body.appendChild(this.popoverRoot);
 
+      this.setPlacementThrottled = throttle(this.setPlacement, 250);
```
Comment (lowiebenoot) [inline components/popover/Popover.js]: Why did it move to `componentDidMount` and why did you change it to 250ms? (16ms is to get approximately 60 fps)

Comment (Kemosabert) [reply components/popover/Popover.js]: I put it in componentDidMount because when I put it outside, it threw an error, saying throttle expected a function. I just checked again, and if I put this line after the setPlacement() function, it does work. What would be the reasoning for not putting it in componentDidMount()?

I don't believe the repositioning calculation needs to happen at 60fps and I believe it might have an impact on performance if you have this happening on pages that also have a lot of things happening on resizing of the window.

Comment (lowiebenoot) [reply components/popover/Popover.js]: In my opinion it's a bit weird to put it in a lifecycle function, as it has nothing to do with the lifecycle. I would just put it after the `setPlacement` function (or in a constructor, but I prefer the first one).

Comment (Kemosabert) [reply components/popover/Popover.js]: That makes sense, I moved the function.

 :ok: by @Kemosabert on 03/07/2018 14:18:32
